Gems/ScriptCanvas/Code/Builder/ScriptCanvasBuilder.cpp

```diff
@@ -425,6 +425,34 @@ namespace ScriptCanvasBuilder
 
         return AZ::Success(result);
     }
+
+    bool ReplaceAsset(ScriptCanvas::RuntimeDataOverrides& overrides, ScriptCanvas::RuntimeAssetPtr asset)
+    {
+        if (!asset.Get())
+        {
+            return false;
+        }
+
+        overrides.m_runtimeAsset = asset;
+        auto dependencyCount = overrides.m_dependencies.size();
+
+        if (dependencyCount == asset->m_runtimeData.m_requiredAssets.size())
+        {
+            for (size_t index = 0; index < dependencyCount; ++index)
+            {
+                if (!ReplaceAsset(overrides.m_dependencies[index], asset->m_runtimeData.m_requiredAssets[index]))
+                {
+                    return false;
+                }
+            }
+        }
+        else
+        {
+            return false;
```

Comment:
Warning/Errors required here? -*Any valid reason this would return false?*- Ignore, I see the code comment on the method in the header :)

Reply:
nitpick: can probably clean the logic a bit with this:

```
if (dependencyCount != asset->m_runtimeData.m_requiredAssets.size())
{
    return false;
}

for (size_t index = 0; index < dependencyCount; ++index)
{
    if (!ReplaceAsset(overrides.m_dependencies[index], asset->m_runtimeData.m_requiredAssets[index]))
    {
        return false;
    }
}

return true;
```